docker_ci_deploy/__main__.py

```diff
@@ -119,37 +176,35 @@ def run(self, images, tags=None, login=None, registry=None):
         :param tags:
             A list of tags to tag the image with or None if no new tags are
             required.
+        :param version:
+            The version to prepend tags with.
         :param login:
             Login details for the Docker registry in the form
             <username>:<password>.
         :param registry:
             The address to the Docker registry host.
         """
         # Build map of images to tags to push with provided tags
-        image_to_push_tags = []
+        tag_map = []
         for image_tag in images:
-            if tags is not None:
-                image_name = strip_image_tag(image_tag)
-                push_tags = ['%s:%s' % (image_name, tag) for tag in tags]
-            else:
-                push_tags = [image_tag]
+            image, tag = split_image_tag(image_tag)
```

Comment:
Would it make sense to factor out the inside of this loop into a separate function?